File: poke-backend/src/main/java/com/ironhack/pokebackend/model/Trainer.java
package com.ironhack.pokebackend.model;

import javax.persistence.*;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;

@Entity
public class Trainer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @NotEmpty
    private String name;
    @Min(0)
    private Integer age;
    @NotEmpty
    private String hobby;
    private String photo;

    public Trainer() {
    }

    public Trainer(String name, Integer age, String hobby, String photo) {
        this.name = name;
        this.age = age;
        this.hobby = hobby;
        this.photo = photo;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getHobby() {
        return hobby;
    }

    public void setHobby(String hobby) {
        this.hobby = hobby;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    @Override
    public String toString() {
        return "Trainer{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", age=" + age +
                ", hobby='" + hobby + '\'' +
                ", photo='" + photo + '\'' +
                '}';
    }
}

File: pokemon-client/src/app/models/Trainer/trainer.ts
export class Trainer {
    
    constructor(
        private _id: number,
        private _name: string,
        private _age: number,
        private _hobby: string,
        private _photo: string
    ) {  
    }

    public get id(): number {
        return this._id;
    }
    
    public get name(): string {
        return this._name;
    }

    public get age(): number {
        return this._age;
    }

    public get hobby(): string {
        return this._hobby;
    }

    public get photo(): string {
        return this._photo;
    }

    public set id(id: number) {
        this._id = id;
    }

    public set name(name: string) {
        this._name = name;
    }

    public set age(age: number) {
        this._age = age;
    }

    public set hobby(hobby: string) {
        this._hobby = hobby;
    }

    public set photo(photo: string) {
        this._photo = photo;
    }
    
}
